docs/reference/api/platform/Debug.md

```diff
@@ -1,11 +1,36 @@
 ## Debug
 
-[Add description here.]
+Mbed OS provides a set of debug functions that you can use to output debug messages to `STDIO` at runtime. `mbed_debug.h` declares these functions, which are available only in debug builds.
 
-### Debug class reference
+### Debug functions reference
 
-[Embed class reference here. If you've never done this before, please ask your editor.]
+[![View code](https://www.mbed.com/embed/?type=library)](https://os.mbed.com/docs/v5.6/mbed-os-api-doxy/group__platform__debug.html)
+
+### Debug functions usage
+
+The `debug` function is a printf-style function that takes a format string followed by arguments, as the example below shows.
+
+```C
+void *operator new(std::size_t count) {
+    void *buffer = malloc(count);
+    if (NULL == buffer) {
+        error("Operator new out of memory\r\n");
+    } else {
+        debug("Operator new succeded");
+    }
+    return buffer;
+}
+```
+
+The `debug_if` function is similar to the `debug` function except that it takes an additional argument as its first parameter. The message prints to `STDIO` only if the first paramater evaluates to `true`.
+
+```C
+void *operator new(std::size_t count) {
+    debug_if( count == 0, "\nnew called with 0 size");
+    ...
+} 
+```
 
 ### Debug example
 
-[Add example here.]
+[![View Example](https://www.mbed.com/embed/?url=https://github.com/ARMmbed/mbed-os-example-platform-utils)](https://github.com/ARMmbed/mbed-os-example-platform-utils) 
```

Comment:
@sg- Should this be in the Mbed example teams folder like the other examples?

Reply:
This example program demonstrates MemoryStats and then have few things to demonstrate debug functions. That's why I have it as a separate example in its own repo. Does that make sense?

Reply:
Hi Sam,
Yeah, it makes more sense to move it to example teams folder. Ill do that, Thanks for pointing that out.

Reply:
Hi Sam, Looks like I need permission to publish to mbed_example. When I try to publish "mbed_example" is not showing as a option for me. Can you please add me to the mbed_example repo?

Reply:
what is your developer account name?